### PR TITLE
Back to Bublé (-250b 🔥)

### PR DESCRIPTION
**Pull request by developit**
- Switch back to Bublé to save a bunch of bytes (1500 -> 1250)
- Convert JSX to plain-old-javascript
- Add a tiny single-purpose `assign()` helper (babel's is large)

/cc @billneff79 @pl12133

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,25 +34,24 @@
...
+  "license": "BSD-3-Clause",
   "files": [
     "src",
     "dist"
   ],
   "eslintConfig": {
-    "extends": "eslint-config-synacor"
+    "extends": "eslint-config-synacor",
+    "rules": {
+      "guard-for-in": 0
```
Comment (billneff79) [inline package.json]: why did you add this here instead of just adding a comment to ignore the rule where it was safe to ignore it?

Comment (developit) [reply package.json]: I believe it was in two places. It's a slightly overzealous rule for library code.

File: src/lib/util.js [hunk 2]
```diff
@@ -6,11 +6,22 @@ export function defined(obj) {
 }
 
 
+/** A simpler Object.assign
+ *  @private
+ */
+export function assign(obj, props) {
```
Comment (billneff79) [inline src/lib/util.js]: This PR looks good.  The only reason why I kept `Object.assign` originally instead of doing a simply polyfill and using `h` instead of spread operators in JSX  was because I was concerned that a year from now when someone comes back and touches this that we would forget that we can't use those.  What are your thoughts on what this does to maintainability?
